cnvlib/reference.py

```diff
@@ -94,9 +93,12 @@ def infer_sexes(cnn_fnames, is_male_reference):
     for fname in cnn_fnames:
         cnarr = read_cna(fname)
         if cnarr:
-            is_xx = cnarr.guess_xx(is_male_reference)
-            if is_xx is not None:
-                sexes[cnarr.sample_id] = is_xx
+            if female_samples:
```

Comment:
Here, `female_samples` can be True (samples are known XX), False (samples are known XY), or None (not specified by user). Untested, but I think this block should be:

    if female_samples is not None:
        sexes[cnarr.sample_id] = female_samples

Reply:
Thanks so much for reviewing this. I don't totally understand how `female_samples` and `male_reference` are meant to interact but my intuition was that `female_samples` being true meant that all samples were female. So the behavior I tried to encode here was to make everything female if true, otherwise to look up on a case by case bases using `is_male_reference` and the chromosome inference. Are all samples male if `female_samples` is False? If so, I can swap over the logic. Thanks again for the help.

Reply:
Yes, both of those are correct. This comes from the command line option `-x (Female|Male)`, where the user can specify that all samples are female (True) or male (False). If it's not used at all, it comes in with the value None, and in that case we guess each sample's chromosomal sex individually. The option is there in case inference is wrong and/or the user knows the sample sexes, e.g. they're studying prostate cancer.

The `is_male_reference` value is just True or False, and it determines whether the reference should have an X ploidy of 1 or 2. If 1 (command line flag `-y`), then the X coverage of "female" input samples is halved, otherwise the X coverage of "male" samples is doubled.

In the prostate cancer, reasonable options would be `-x Male -y` to specify that all input samples are male (`-x Male`), and that the output reference should have an X ploidy of 1 (`-y`). For ovarian cancer you'd use `-x Female` by itself, and if you're used to array CGH, you'd use `-y` by itself and let inference happen. Generally, users ignore these options until they need them. The same two options appear in several other commands and have the same effect, but the terminology is most confusing in `reference`.

Anyway, I thought I'd written this so that if `female_samples` is specified (True or False, not None), the initial pass of loading .cnn files to infer sex is skipped, and later if the `sexes` dictionary then empty, a sample's sex is the value of `female_samples` -- maybe I missed a `sexes.get(sample_id, female_samples)` somewhere.